src/inspection/src/inspection_ros.cpp

```diff
@@ -0,0 +1,71 @@
+#include "include/inspection_ros.hpp"
+
+InspectionMission::InspectionMission() : Node("inspection") {
+  // creates publisher that should yield torque/acceleration/...
+  control_command_publisher =
+  this->create_publisher<fs_msgs::msg::ControlCommand>("/control_command", 10);
+
+  // creates publisher for the flag
+  finish_publisher = this->create_publisher<fs_msgs::msg::FinishedSignal>("/signal/finished", 10);
+
+  // get mission
+  mission_subscription = this->create_subscription<fs_msgs::msg::GoSignal>(
+      "/wheel_states", 10,
+      std::bind(&InspectionMission::mission_decider, this, std::placeholders::_1));
+}
+
+void InspectionMission::mission_decider(fs_msgs::msg::GoSignal mission_subscription) {
```

Comment:
I would change the name of the variable, as this is the actual message you are receiving, not a subscription